extensions/aggregate-terms.js

```diff
@@ -27,32 +35,91 @@ module.exports.register = function ({ config }) {
     return termContent;
   }
 
+  /**
+   * Load terms from a specified local path if provided.
+   */
+  function loadLocalTerms(siteCatalog, termsPath) {
+    try {
+      const resolvedPath = path.resolve(termsPath);
+      if (!fs.existsSync(resolvedPath)) {
+        logger.warn(`Local terms path "${termsPath}" does not exist.`);
+        return false;
+      }
+
+      const termFiles = fs.readdirSync(resolvedPath).filter(file => file.endsWith('.adoc'));
+      if (!termFiles.length) {
+        logger.warn(`No term files found in local path "${termsPath}".`);
+        return false;
+      }
+
+      termFiles.forEach(file => {
+        const filePath = path.join(resolvedPath, file);
+        const termContent = fs.readFileSync(filePath, 'utf8');
+        const categoryMatch = /:category: (.*)/.exec(termContent);
+        const category = categoryMatch ? categoryMatch[1] : 'Miscellaneous';
+
+        const formattedCategory = category.charAt(0).toUpperCase() + category.slice(1);
```

Comment:
We could consider adding this capitalize function in an util file 
https://github.com/redpanda-data/docs-extensions-and-macros/blob/main/macros/rp-connect-components.js#L49
https://github.com/redpanda-data/docs-extensions-and-macros/tree/main/extensions/util